### PR TITLE
Fixed a "can't get key of undefined problem" and added error handling logic to oracle page

### PR DESCRIPTION
**Pull request by chrisli30**

### CODE REVIEW

File: src/containers/oracle.js [hunk 1]
```diff
@@ -134,6 +135,7 @@ class OraclePage extends React.Component {
               // Finalize oracle if current block has passed arbitrationEndBlock and threshold is not met
               // Since new oracle is guarranteed to spawn if total amount threshold is met we are not checking total BOT amount here
               if (blockCount > oracle.endBlock) {
+                console.log(`!! blockCount ${blockCount} is greater than endBlock ${endBlock}.`);
                 configName = 'FINALIZING';
```
Comment (chrisli30) [inline src/containers/oracle.js]: console.log here for debugging purpose. Will clear them up later

File: src/containers/oracle.js [hunk 2]
```diff
@@ -148,9 +150,14 @@ class OraclePage extends React.Component {
             configName = 'SETTING';
             break;
           default:
+            console.warn('Oracle exists but cant determine status. ');
+            oracle = undefined;
             break;
```
Comment (chrisli30) [inline src/containers/oracle.js]: Basically, if we can't identify an oracle's status with above rules, we are gonna show 404 in the center of page, to which I haven't added style yet.

File: src/containers/oracle.js [hunk 3]
```diff
@@ -201,16 +207,18 @@ class OraclePage extends React.Component {
...
+          console.log('onSetResult');
           onSetResult(contractAddress, selectedIndex, senderAddress);
         }, SUB_REQ_DELAY);
         break;
 
       case 'VOTING':
 
         /** The amount of voting needs to be approved by Bodhi_token * */
-        onApprove(oracle.topicAddress, amount, senderAddress);
+        onApprove(_.trimStart(oracle.address, '0x'), amount, senderAddress);
```
Comment (chrisli30) [inline src/containers/oracle.js]: Every address's "0x" needs to be trimmed before send to api. I am open to suggestions.

Comment (dwalintukan) [reply src/containers/oracle.js]: Already fixed in API/qweb3 layer. No need to trim any addresses of any sort of the `0x`.

File: src/containers/oracle.js (hunk 2 above)
Comment (dwalintukan) [reply src/containers/oracle.js]: 👍

File: src/containers/oracle.js (hunk 1 above)
Comment (dwalintukan) [reply src/containers/oracle.js]: 👍

**Pull request merged**